src/components/BookCard/BookCard.tsx

```diff
@@ -1,13 +1,44 @@
-import type { Book } from '../../domain/books/types'
+import { useState } from "react";
+import { BooksPage } from "../../domain/books/BooksPage";
+import type { Book } from "../../domain/books/types";
+import { CartPage } from "../../domain/cart/CartPage";
+import { HashRouter as Router, Route, Link, Routes } from "react-router-dom";
 
 /* 
   We would define the structure of the props in the same file as the component.
   Since, this component has the same props as Type Book, we would assign it to it props.
 */
-type BookProps = Book;
+// interface BookProps {
+//   books: Book[];
+// }
+
+export const BookCard = (): JSX.Element => {
+  const [bookData, setBookData] = useState<Book[]>([]);
+
+  const handleClick = (e: React.MouseEvent, book: Book) => {
+    // console.log("hello there!!!:", book);
+    setBookData([...bookData, book]);
+  };
```

Comment:
1) We are using Redux-Toolkit, all of state logic should be handled there.
2) Why is BookCard a top level component?